vortex-array/src/validity.rs

```diff
@@ -208,8 +208,13 @@ impl Validity {
 
     pub fn take(&self, indices: &Array) -> VortexResult<Self> {
```

Comment:
Since we started putting nulls in dictionary codes array we have to handle take indices being nullable

Reply:
add tests